src/camera/Camera.cpp

```diff
@@ -23,7 +24,24 @@ bool Camera::open(int camera_id, CameraParams intrinsic_params, Mat extrinsic_pa
 		return false;
 	}
 	_capture_lock->lock();
-	bool result = this->_capture->open(camera_id);
+	std::stringstream gstr_ss;
+	gstr_ss << "v4l2src device=/dev/video" << camera_id << " ! ";
+	if (camera_id == 20) {
+		gstr_ss << "image/jpeg,width=640,height=480,framerate=30/1 ! "
+				   "jpegdec ! videoconvert ! appsink";
+	} else if (camera_id == 40 || camera_id == 30) {
+		gstr_ss << "image/jpeg,width=1024,height=768,framerate=30/1 ! "
+				   "jpegdec ! videoconvert ! appsink";
+	} else if (camera_id == 100) {
+		gstr_ss << "image/jpeg,width=1280,height=720,framerate=30/1 ! "
+				   "jpegdec ! videoflip method=rotate-180 ! videoconvert ! appsink";
+	} else if (camera_id == 110) {
+		gstr_ss << "image/jpeg,width=960,height=720,framerate=30/1 ! "
+				   "jpegdec ! videoflip method=rotate-180 ! videoconvert ! appsink";
+	}
+	LOG_F(INFO, "GSTR: %s", gstr_ss.str().c_str());
+	this->_capture = std::make_shared<cv::VideoCapture>(gstr_ss.str(), cv::CAP_GSTREAMER);
+	bool result = this->_capture->isOpened();
```

Comment:
OpenCV on the jetson uses the GStreamer backend by default, which does not respect some parameter configurations such as `fourcc`, which is used to set the encoding. (So it was defaulting to yuyv instead of mjpg, which is what we were setting it to)

This code constructs [gstreamer pipelines](https://gstreamer.freedesktop.org/documentation/tools/gst-launch.html?gi-language=c) ([this ](https://gist.github.com/hum4n0id/cda96fb07a34300cdb2c0e314c14df0a) might be of some use) which set the resolution, framerate, and encoding as necessary, which are actually respected by GStreamer.

Note that you have to set the resolution/framerate/encoding to a combination actually supported by the device, which you can find by following [this](https://askubuntu.com/a/687140/1664152).

Observe how we can also do basic transformations, such as rotating the science cameras (id 100 and 110) by 180 degrees, since those cameras were mounted upside down. We can also do stuff like downsample framerate (which could be useful, for example, if a stream is available at 60fps but we only want 30fps)

The camera config files should be overhauled to support this format of specification, which is a lot better imo. I think we should also probably get rid of the `Camera::open()` method that accepts a filename, since it adds extra maintenance overhead and I don't think we'll be using it.

Reply:
Refactor as necessary to support these changes